native/utils/clear-platform.js

```diff
@@ -2,43 +2,82 @@
 
 const chalk = require('chalk');
 const path = require("path");
-const { existsSync, removeSync } = require('fs-extra');
+const fs = require('fs-extra');
 
-// npm run clear-platform
+// usage: npm run clear-platform
 let externalDir = path.join(__dirname, "../external");
 let failed = false;
 
+let macUsefulDirs = ['android', 'cmake', 'emscripten', 'ios', 'ios-m1-simulator', 'mac', 'ohos', 'sources'];
+let winUsefulDirs = ['android', 'cmake', 'emscripten', 'ohos', 'sources', 'win32', 'win64'];
+let linuxUsefulDirs = ['android', 'cmake', 'emscripten', 'linux', 'ohos', 'sources'];
+let allDirs = [];
+
 main();
 
 function main() {
-    clearUselessPlatform();
+    fs.readdir(externalDir, function(err, files){
+        (function iterator(i) {
+          if(i == files.length) {
+            console.log(chalk.green(`==== current exists 3rd-libs directories ====`));
+            console.log(allDirs);
+            cleanPlatform(process.platform);
+            return;
+          }
+          fs.stat(path.join(externalDir, files[i]), function(err, data) {     
+            if(data.isDirectory() && files[i].charAt(0) !== '.') {               
+                allDirs.push(files[i]);
+            }
+            iterator(i+1);
+           });   
+        })(0);
+    });
 }
 
-function clearUselessPlatform() {
-    console.log(chalk.magenta(`==== Remove useless 3rd-libs for current platform ====`));
-    console.log(`  ${chalk.green('current platform: ')} ${process.platform}`);
-    console.log(`  ${chalk.green('3rd-libs directory: ')} ${externalDir}`);
-    if (process.platform === 'darwin') {
-        // remove win32 related 3rd-libs
-        console.log(chalk.magenta(`==== Remove win32 related 3rd-libs ====`));
-        let clearDirectory = path.join(externalDir, "./win32");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
-        clearDirectory = path.join(externalDir, "./win64");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
-    } else {
-        // remove apple related 3rd-libs
-        console.log(chalk.magenta(`==== Remove apple related 3rd-libs ====`));
-        let clearDirectory = path.join(externalDir, "./mac");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
-        clearDirectory = path.join(externalDir, "./ios");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
-        clearDirectory = path.join(externalDir, "./ios-m1-simulator");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
+function cleanPlatform(platform) {
+    if (platform === 'darwin') { // macOS
+        let macUselessDirs = getArrDiff(allDirs, macUsefulDirs);
```

Comment:
Why not just use `macUsefulDirs ` as `allDirs` is empty.

Reply:
`allDirs` will get all dirs in external firstly in main function.